web-wallet/src/lib/stores/mnemonicPhraseResetStore.js

```diff
@@ -0,0 +1,6 @@
+import { writable } from "svelte/store";
+
+/** @type {import('svelte/store').Writable<string[]>} */
+const mnemonicPhraseResetStore = writable([]);
```

Comment:
The problem here is that you're putting the mnemonic in memory, while we carefully avoided to do so until now.

Reply:
So what other options do we have?

Reply:
Hmm, i want to ask a question as I might have an idea for this. Are you saving the mnemonic in a store just so that you have it available for the restore process?

Reply:
Yeah... the idea is that if the user tries to enter a mnemonic on the login page, and it's a different than the Wallet they have stored locally, we should instead re-direct the user to the restore flow (as it will add the new step, to allow to sync from a specific block), but it's also nice to have unified experience, I think this makes more sense.

So yeah, the idea is the user would have already entered the new mnemonic on the login page (and it will be a valid one, as it's validated, otherwise the restore flow won't be triggered at all), so they don't need to re-enter it again on the restore flow. Another option would be to completely skip this step, but still we have to keep it somewhere until we initialize the wallet.

Any ideas are welcome and appreciated.

Reply:
I understand that you want to implement something that is for QoL, but we are trading security for it. Given that the user will most likely paste the mnemonic he can do so again in the restore flow. This case is only if he does not go directly to the restore flow via the button present on that page.

Because data has to be passed between pages I do not see any other option than a store without refactoring the whole restore and login pages.

A very dramatic solution would be to merge the login and restore flow into the same page and use events to pass the data around, thus eliminating the storage of the mnemonic.

Reply:
I would involve @ZER0 for his input on the matter.

Generally, the experience I have implemented is what we discussed with him yesterday. I think that, by copying the mnemonic in the clipboard, the user is already taking some risk (this is not to say that we shouldn't try to minimize it, whenever possible).

As you said, it will always be a trade-off between usability and security. I guess the question is what the overall risk is (severity * probability) against the benefits users will experience. If the benefits outweigh a low risk, it might be worth pursuing.

However, as I'm not an expert in memory management or related attacks, I can't assess the risk of temporarily storing the mnemonic in a Svelte store. 

Pre-populating the mnemonic into the view may not significantly enhance user experience either, as users would have already copied the mnemonic, as you have mentioned, and could be perceived either as a quality or a security oversight, based on the implementation choice.

Having said that, I am happy to proceed in whichever of the ways we collectively agree on.

Reply:
It was decided to show the screen with the mnemonic displayed, as implemented here.